eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ga4gh/beaconv2/BeaconServiceV2.java

```diff
@@ -298,7 +300,7 @@ private BeaconAlleleRequest buildBeaconAlleleRequest(String chromosome, Long sta
     private BeaconDatasetAlleleResponse buildDatasetAlleleResponseHelper(boolean exists, VariantSource variantSource,
                                                                          Float frequency) {
         return new BeaconDatasetAlleleResponse()
-                .datasetId(variantSource.getStudyId())
+                .datasetId(variantSource.getStudyId() + "_" + variantSource.getFileId())
```

Comment:
`variantSource.getStudyId() + "_" + variantSource.getFileId()`: although sometimes we extract these kind of things to a function. I'm not convinced this time is worth it.